pkgs/development/python-modules/tifffile/default.nix

```diff
@@ -22,12 +23,12 @@ buildPythonPackage rec {
     pytest
   '';
 
-  propagatedBuildInputs = [ numpy ]
+  propagatedBuildInputs = [ numpy imagecodecs-lite ]
     ++ lib.optional isPy27 [ futures enum34 pathlib ];
```

Comment:
The same applies to the patch above.

Reply:
please leave it in for now. It's not unlikely that someone would choose to re-add Python 2 support.

Reply:
I modified so imagecodecs-lite is only loaded for python3..let me know if this sounds good?

Reply:
what does the `setup.py` say?

Reply:
According to README on PYPI, imagecodes is optional (https://pypi.org/project/tifffile/). Here's setup.py:

```
install_requires=[
        'numpy>=1.11.3',
        'pathlib;python_version=="2.7"',
        'enum34;python_version=="2.7"',
        'futures;python_version=="2.7"',
        # require imagecodecs on Windows only
        imagecodecs + ';platform_system=="Windows"',
    ],
    extras_require={
        'all': ['matplotlib>=2.2', imagecodecs],
    },
    tests_require=['pytest', imagecodecs,
                   'czifile', 'cmapfile', 'oiffile', 'lfdfiles'],
```

Reply:
so then we do *not* include it. Users can add it themselves if they want to.

Reply:
ah ok. Should this be done by users with overlays, e.g.

````tifffile = python-super.tifffile.overrideAttrs  (oldAttrs: rec {
      buildInputs = oldAttrs.buildInputs ++  [ python-super.imagecodecs-lite ];
    });```

Or should I add a flag to the derivation?

Reply:
no, when you want to use them, then you would just add them to the environment. Generally python scripts are used within a shell.

example shell.nix
```
let
  pkgs = import <nixpkgs> {};
in

with pkgs;
with python37Packages;
mkShell {
  name = "pip-env";
  buildInputs = [
    virtualenv
    tifffle
    matplotlib
  ];
}
```

Reply:
ah I see, ty!

Reply:
@jonringer when you want to have an environment that represents a build environment, that works well. Otherwise, I would recommend going of the `nix run` approach so setup hooks are not executed.

For example, having the following in a `default.nix` or `shell.nix` works when using `nix-shell`
```
with import <nixpkgs> {};
(python3.withPackages(ps: with ps; [ virtualenv tifffle matplotlib imagecodecs-lite ]).env
```
Note the env. When using `nix run -f . ` that part should be removed
```
with import <nixpkgs> {};
(python3.withPackages(ps: with ps; [ virtualenv tifffle matplotlib imagecodecs-lite ])
```

Reply:
in general that is my workflow. I guess I could do a default.nix like the on you have, then I could have just have a shell.nix be:
```
  (import ./default.nix).env
```
I think that would work...

Reply:
or just execute `nix run -f .` :)

Reply:
too easy, don't like :)

Reply:
Also for my workflow, i need to be able to use virtualenv and then install some packages locally, the shellhook is nice in that case, because i can have it auto create the venv dir, set the pythonpath to include the venv sitepackages, and a few other project specific things :)